core/server/api/shared/validators/input/all.js

```diff
@@ -118,14 +121,31 @@ module.exports = {
         const jsonpath = require('jsonpath');
 
         if (apiConfig.data) {
-            const missedDataProperties = _.filter(apiConfig.data, (value, key) => {
-                return jsonpath.query(frame.data[apiConfig.docName][0], key).length === 0;
+            const missedDataProperties = [];
+            const nilDataProperties = [];
+
+            _.each(apiConfig.data, (value, key) => {
+                if (jsonpath.query(frame.data[apiConfig.docName][0], key).length === 0) {
+                    missedDataProperties.push(key);
+                } else if (_.isNil(frame.data[apiConfig.docName][0][key])) {
+                    nilDataProperties.push(key);
+                }
             });
 
             if (missedDataProperties.length) {
                 return Promise.reject(new common.errors.ValidationError({
                     message: common.i18n.t('notices.data.validation.index.validationFailed', {
-                        validationName: 'FieldIsRequired'
+                        validationName: 'FieldIsRequired',
+                        key: JSON.stringify(missedDataProperties)
+                    })
+                }));
+            }
+
+            if (nilDataProperties.length) {
+                return Promise.reject(new common.errors.ValidationError({
+                    message: common.i18n.t('notices.data.validation.index.validationFailed', {
+                        validationName: 'FieldIsInvalid',
+                        key: JSON.stringify(nilDataProperties)
```

Comment:
```suggestion
                    key: key
```
As `missedDataProperties` is an array, would it be more explicit naming this property `keys` instead :thinking: ? Similar goes to the other `key` usages in validations.

Reply:
We use the `notices.data.validation.index.validationFailed` key for many use cases in Ghost. And it's not always an array. You can push whatever value you want.

hmmm i was thinking about this too a little, but ignored it at the end, because i would need to handle single matches and multiple matches? I thought this is not important at the moment 🤠

Reply:
Agreed. In the end of the day it's just an 'additional info', so not super critical to be 100% conventional here :+1:

Reply:
Your spot was 100% correct, this is not suuuuper smart code. Let's keep this in mind :)